packages/panzer/disc-fe/src/Panzer_Evaluator_DomainInterface.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef PANZER_EVALUATOR_DOMAIN_INTERFACE_HPP
+#define PANZER_EVALUATOR_DOMAIN_INTERFACE_HPP
+
+namespace panzer {
+
+  class Workset;
+
+  /** \brief Mix-in interface to support cell "domains" in panzer.
+   * 
+   *  This class adds support for cell domains into evaluators. This is commonly used in DG discretizations.
+   */
+  class DomainEvaluator {
+
+  public:
+
+    /// Domain types supported by worksets. 
+    enum : int {
+        OWNED=0,    /// All Owned cells for the workset on the MPI process
+        GHOST=1,    /// All Ghosted cells for the workset on the MPI process
+        REAL=2,     /// All Owned and Ghosted cells for the workset on the MPI process
+        VIRTUAL=3,  /// All virtual cells for the workset on the MPI process
+        ALL=4       /// All OWNED, GHOSTED and VIRTUAL cells for the workset on the MPI process
+        };
+
+    /**
+     * \brief Constructor
+     *
+     * \param[in] domain (optional) Cell domain to iterate over
+     */
+    DomainEvaluator(int domain=ALL);
```

Comment:
Why don't you give a name to the enum? It would be a bit more descriptive. Anyway, I fine with this implementation if you think it is best.

Reply:
I would like this as well.

Reply:
Funny. I originally had a name there, but found that the current empire implementation did not have a name, and removed it to be consistent with empire. Will add it back.